tests/publiccloud/sles4sap.pm

```diff
@@ -27,15 +37,187 @@ sub upload_ha_sap_logs {
     }
 }
 
+=head2 wait_until_resources_started
+
+    wait_until_resources_started();
+
+Wait for HA cluster to be started.
+=cut
+sub wait_until_resources_started {
```

Comment:
It's too bad this is partially duplicating `hacluster::wait_until_resources_started()`. Ideally, this one should be something similar to a wrapper around the other one, handling cloud-only stuff, and then running `hacluster::wait_until_resources_started()`, but the differences between `script_run` and `$self->{my_instance}->run_ssh_command()` make it difficult to refactor a single function for both.

Reply:
Yes, maybe we can think of a FULL refactor on the way we are currently doing SAP+HA tests on PC, to act more current openQA tests. But this will be a long time option I think, and I would like to finally have workable PC tests :-)

Reply:
So for later improvements, I will recommend you the stuff from @pdostal. He basically is using a console which is connected via SSH to the PC instance. And then just use `script_run()` and so.
But I see difficulties with multi-VM scenarios here.

My quickest solution would be passing a function pointer to `hacluster::wait_until_resources_started()`. function like:
```
# in sles4sap.pm
sub wait_until_resources_started {
  my %args = @_;
  $args{script_run} = sub { return $instance->run_ssh_cmd(cmd=>shift, timeout=>shift, rc_only => 1);};
  return hacluster::wait_until_resources_started(%args);
}
```
and in `hacluster::wait_until_resources_started()`
```
# in hacluster.pm
sub wait_until_resources_started {
  my %args = @_;
  $args{script_run} //= \&script_run;
  ...
  while ($ret = $args{script_run}->("$cmd", $default_timeout)) {
  ...
}
```

just an idea, not sure if it makes sense...
Actually it would be nice to transparently overwrite `script_run()` but I didn't found something workable for me.

Reply:
> Yes, maybe we can think of a FULL refactor on the way we are currently doing SAP+HA tests on PC, to act more current openQA tests. But this will be a long time option I think, and I would like to finally have workable PC tests :-)

I agree with you about the long time option and that you would like a quick solution :smile: 
Maybe we could give a look at the full refactor after the release of 15-SP2. Moreover, I think it could be a good topic for Christian.

Reply:
> My quickest solution would be passing a function pointer to `hacluster::wait_until_resources_started()`. function like:

Yes, I was also thinking about passing a "script_run()"-like function as an argument to `hacluster::wait_until_resources_started()`, but we'll have to keep in mind several regression scenarios, so better to wait for the moment.

> just an idea, not sure if it makes sense...

Perfect sense to me.